### DISCUSSION TITLE
How to concatenate multiple yaml files into an array

### DISCUSSION THREAD

**Discussion by chevdor** [mikefarah/yq#1426] [General]
`a.yaml`:
```
n: 1
```

`b.yaml`:
```
n: 2
```

I would need the output ot be as shown below and the solution to work with `*.yaml` as input:
```
[
   { n: 1 },
   { n: 2 }
]
```

or the yaml equivalent.

**Comment by chevdor** [comment]
This option only works for 2 files. I should have mentioned that I need to process `*.yaml` based on the example above.
Edit: Question fixed.

**Comment by joao-paulo-parity** [comment]
For 2 files:

`jq --arg first "$(yq -o=json eval a.yml)" --arg second "$(yq -o=json eval b.yml)" -n '[$first | fromjson] + [$second | fromjson]'`

For a variable amount of files:

```bash
#!/bin/bash

args=()

expr=""
arg_i=0
for file in "$@"; do
  args+=(--arg "arg${arg_i}" "$(yq -o=json eval "$file")")
  expr+=" [\$arg${arg_i} | fromjson] +"
  arg_i=$(( arg_i + 1 ))
done
expr="${expr:0: $(( ${#expr} - 2 ))}"

jq "${args[@]}" -n "$expr"
```

**Reply by chevdor** [reply]
Extra points is tihs is using `yq` only :)

**Comment by mikefarah** [comment]
```bash
yq ea '[.]' data1.yaml data2.yaml
```

Explanation:
- use `ea` to execute against all the yaml content at once (by default, yq runs against one document at a time)
- surround everything by an array

**Reply by mikefarah** [reply]
Though this doesn't have the format you had, are you trying to get JSON? You can use the `-o=j` flag to get that layout

**Reply by chevdor** [reply]
Perfect, this is exaclty what I needed ! Thank you.
To summarize, json was indeed what I was after and it works fine with `yq ea -o=j '[.]' *.yaml`.
